command/FaviconHelper.js

```diff
@@ -0,0 +1,86 @@
+const Database = require('../data/database');
+const database = new Database();
+const url = require('url');
+const getFavicons = require('get-website-favicon');
+const scrape = require('html-metadata');
+
+class FaviconHelper {
+  constructor() {
+    this.currentRequests = {};
+  }
+
+  clearCurrentRequests() {
+    this.currentRequests = {};
+  }
+
+  getHostname(link) {
+    return url.parse(link).hostname.replace('www.', '');
+  }
+
+  async getWebsiteFavicon(link) {
+    let hostname = this.getHostname(link);
+    let data = await database.getWebsiteFavicon(hostname);
+    let expire_date = new Date();
+    // Set the expire_date to 30 days ago
+    expire_date.setDate(expire_date.getDate() - 30);
+
+    if (data && new Date(data.updated_on) > expire_date) {
+      return data;
+    } else {
+      let self = this;
+      if (!self.currentRequests[hostname]) {
+        let faviconPromise = getFavicons(link);
+        let metadataPromise = scrape(link);
```

Comment:
So I should make two separate tables (and two helpers FaviconHelper, PublisherHelper) : one linking the hostname to the favicon and one linking the hostname to the publisher name?

Reply:
No, it makes sense to have them in one table. I would just make this function more flexible so that you can retrieve either or both types of data depending on what client is asking for. The AudioFileLocations data does something similar for handling the full-audio & summary files.

Reply:
Got it

Reply:
Done, parameter type to specify between 'all', 'favicon' and 'publisher'. Defaults to 'all'.
Moved the fetching to another function, also depending on type and updated currentRequests var to reflect this